### ISSUE TITLE
Add note to README about passage id requirements

### ISSUE THREAD

**Issue by fschlatt** [stanford-futuredata/ColBERT#127]
It might be helpful to add a note in the data section of the README that the passage corpus file needs to have ids starting at 0, are sorted and increase by 1. i.e. 

```
0\tpassage1
1\tpassage2
...
n\tpassagen+1
```

This is mostly the case, because the corpora loader completely ignores passage ids from the tsv file and just uses the line index as the id. https://github.com/stanford-futuredata/ColBERT/blob/6914396f4dcd3d5ec33f7a82c13e4af67e82b1d0/colbert/evaluation/loaders.py#L161

I had a look at changing the code such that arbitrary passage ids can be used. This is easily possible for training a model, but becomes more involved for indexing and retrieval. It may be a good addition for the future though. ;)

**Comment by yuval** [comment]
Good point. There's an assert in the same file that fails if the passage ID is not the same as the line index. If you comment it out then things work fine. I also tried to use the line index as the passage ID and putting the real passage ID in the third column. That works but the code uses it to create the passage title and then uses it as a prefix to the passage text as follows: "title + ' | ' + passage". It's not documented but good to know.

**Comment by thibault-formal** [comment]
hello,
I encountered the same issue recently (and fixed it by commenting as you did @yuval), but I was surprised this repository does not support indexing documents with arbitrary `ids` off-the-shelf (which happens surely as soon as you do not use MS MARCO). 🤔

**Comment by fschlatt** [comment]
My fork supports indexing and searching with arbitrary doc ids ;)
